zebra/zapi_msg.c

```diff
@@ -694,10 +696,11 @@ static int zsend_ipv4_nexthop_lookup_mrib(struct zserv *client,
 		 * nexthop we are looking up. Therefore, we will just iterate
 		 * over the top chain of nexthops.
 		 */
-		for (nexthop = re->nhe->nhg.nexthop; nexthop;
```

Comment:
The comment just above here is a bit confusing, since it claims there will be no recursive nexthops?

Reply:
It means that this code will only return/resolve to next hops that are final (non recursive). This PR came as result of a recent discussion in the PIM channel about a problem I was having.

Reply:
Well, but the comment says "just iterate over the top chain", and that is no longer accurate - that's why I was asking about it?

Reply:
fixed